src/app/domain/authentication/services/SynchronizeService.ts

```diff
@@ -1,14 +1,35 @@
 import Minilog from '@cozy/minilog'
 
 import type CozyClient from 'cozy-client'
+import { getErrorMessage } from 'cozy-intent'
 
-const log = Minilog('SynchronizeService')
+import { oauthSaveResponse } from '/app/domain/authentication/models/Device'
+import { translation } from '/locales'
+
+export const syncLog = Minilog('📱 SynchronizeService')
 
 export const synchronizeDevice = async (client: CozyClient): Promise<void> => {
   try {
     await client.getStackClient().fetchJSON('POST', '/settings/synchronized')
-    log.info('📱 Device synchronized')
+    syncLog.info('Device synchronized')
   } catch (error) {
-    log.warn('Error while synchronizing device', error)
+    syncLog.warn('Error while synchronizing device', error)
+  }
+}
+
+export const checkClientName = async (client: CozyClient): Promise<void> => {
+  const clientName = client.getStackClient().oauthOptions.clientName
+
+  if (!clientName.startsWith(translation.software.name)) {
+    try {
+      const { data } = (await client.save({
+        _type: 'io.cozy.oauth.client',
+        name: translation.software.name
```

Comment:
mistake here, the devicename should be added back

Reply:
To prevent this we should extract a method that generates the client's name and use this method everywhere we set the client's name